File: toolloop-back/src/main/java/com/toolloop/model/dto/ToolMapItem.java
package com.toolloop.model.dto;

import com.toolloop.model.entity.Category;
import com.toolloop.model.entity.ToolPhoto;
import com.toolloop.model.entity.User;
import io.quarkus.runtime.annotations.RegisterForReflection;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.List;

@RegisterForReflection
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ToolMapItem {
    public Long toolId;
    public String name;
    public BigDecimal pricePerDay;
    public Boolean isReserved;
    public List<ToolPhoto> photos;
    public Category category;
    public User owner;
    public BigDecimal latitude;
    public BigDecimal longitude;
    public Integer distanceMeters;
}

File: toolloop-front/src/app/core/models/dto/tool-map-item.ts
import { Category } from '../entity/category';
import { ToolPhoto } from '../entity/tool-photo';

export interface ToolMapItem {
    toolId: number;
    name: string;
    pricePerDay: number;
    isReserved: boolean;
    photos: ToolPhoto[];
    category: Category;
    owner: { name: string; averageRating: number };
    latitude: number;
    longitude: number;
    distanceMeters?: number;
}
